airflow/jobs/scheduler_job.py

```diff
@@ -783,6 +783,22 @@ def _execute(self) -> None:
                     self.log.exception("Exception when executing DagFileProcessorAgent.end")
             self.log.info("Exited execute loop")
 
+    def _update_dag_run_state_for_paused_dags(self):
+        try:
+            paused_dag_ids = DagModel.get_all_paused_dag_ids()
+            for dag_id in paused_dag_ids:
```

Comment:
Hmm, for a deployment with 3000 dags, I wonder if there would be a challenge here?

Reply:
It's possible that at the first check of the `dag_run`, the `dag_run` state could not be updated because the task instances are still running. If we cache the dag, and the task instances of the `dag_run` `complete` later on, we won't be able to update the `dagrun` state, right?